pkg/collections.go

```diff
@@ -260,38 +288,69 @@ func (c *copier) copyIndexes(ctx context.Context, sourceColl driver.Collection,
 	return nil
 }
 
-// ensureDestinationCollection ensures that a collection exists for a database on the destination and returns it.
-func (c *copier) ensureDestinationCollection(ctx context.Context, db driver.Database, coll driver.Collection, props driver.CollectionProperties) (driver.Collection, error) {
+// createCollection creates a collection on the destination database.
+func (c *copier) createCollection(ctx context.Context, db driver.Database, coll driver.Collection, props driver.CollectionProperties) error {
+	if props.IsSystem {
+		// skip system collections
+		return nil
+	}
 	exists, err := db.CollectionExists(ctx, coll.Name())
 	if err != nil {
 		c.Logger.Warn().Err(err).Msg("Failed to get if collection exists.")
-		return nil, err
+		return err
 	}
-	if !exists {
-		options := &driver.CreateCollectionOptions{
-			JournalSize:       int(props.JournalSize),
-			ReplicationFactor: props.ReplicationFactor,
-			WriteConcern:      props.WriteConcern,
-			WaitForSync:       props.WaitForSync,
-			DoCompact:         &props.DoCompact,
-			CacheEnabled:      &props.CacheEnabled,
-			ShardKeys:         props.ShardKeys,
-			NumberOfShards:    props.NumberOfShards,
-			IsSystem:          false,
-			Type:              props.Type,
-			KeyOptions: &driver.CollectionKeyOptions{
-				AllowUserKeys: props.KeyOptions.AllowUserKeys,
-				Type:          props.KeyOptions.Type,
-			},
-			DistributeShardsLike: props.DistributeShardsLike,
-			IsSmart:              false,
-			ShardingStrategy:     props.ShardingStrategy,
-		}
-		if props.SmartJoinAttribute != "" {
-			options.IsSmart = true
-			options.SmartJoinAttribute = props.SmartJoinAttribute
+	if exists {
+		return nil
+	}
+	options := &driver.CreateCollectionOptions{
+		JournalSize:       int(props.JournalSize),
+		ReplicationFactor: props.ReplicationFactor,
+		WriteConcern:      props.WriteConcern,
+		WaitForSync:       props.WaitForSync,
+		DoCompact:         &props.DoCompact,
+		CacheEnabled:      &props.CacheEnabled,
+		ShardKeys:         props.ShardKeys,
+		NumberOfShards:    props.NumberOfShards,
+		IsSystem:          false,
+		Type:              props.Type,
+		KeyOptions: &driver.CollectionKeyOptions{
+			AllowUserKeys: props.KeyOptions.AllowUserKeys,
+			Type:          props.KeyOptions.Type,
+		},
+		DistributeShardsLike: props.DistributeShardsLike,
+		IsSmart:              false,
```

Comment:
Shoudn't we copy this?

Reply:
Actually, looking at go-driver, there seems to be something wrong. The API does not return isSmart. But it also does not return SmartJoinAttribute it returns SmartGraphAttribute ( for a collection! ). I need to talk to Adam to verify if this is a bug in go-driver or a bug in arango-api.

Reply:
Ok, so two things. The API doc is wrong... Second, IsSmart is actually not used in go-driver... :) Also, not returned by the API at all so it's just an ignored field basically in the create Options.

I'm still setting smartJoinbAttribute though. I'll talk with Adam if we should remove it all together.